src/libraries/System.Runtime.Serialization.Schema/tests/System/Runtime/Serialization/Schema/Import/ImporterTests.cs

```diff
@@ -167,6 +167,51 @@ public static IEnumerable<object[]> Import_MemberData()
                 (XmlSchemaSet schemaSet, XmlQualifiedName typeName) = PrepareFormatVersioningTest(fvArg.type, fvArg.xpath, fvArg.xmlFrag);
                 yield return new object[] { (XsdDataContractImporter imp) => imp.Import(schemaSet, typeName) };
             }
+
+            // DateTime / DateOnly / TimeOnly primitive schema import verification
+            yield return new object[] { (XsdDataContractImporter imp) => {
+                // Construct a schema set with a complex type containing ms:dateOnly & ms:timeOnly plus an xsd:time element
+                string serNs = "http://schemas.microsoft.com/2003/10/Serialization/";
+                string testNs = "http://schemas.datacontract.org/2004/07/DateOnlyTimeOnlyImport";
+                string schema = $"""
+                    <?xml version='1.0'?>
+                    <xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema' targetNamespace='{testNs}' elementFormDefault='qualified'>
+                        <xs:import namespace='{serNs}' />
+                        <xs:complexType name='DateOnlyTimeOnlyContainer'>
+                        <xs:sequence>
+                            <xs:element name='DateOnlyValue' type='ser:dateOnly' xmlns:ser='{serNs}' />
+                            <xs:element name='TimeOnlyValue' type='ser:timeOnly' xmlns:ser='{serNs}' />
+                            <xs:element name='XsdTimeValue' type='xs:time' />
+                            <xs:element name='XsdDateTimeValue' type='xs:dateTime' />
+                        </xs:sequence>
+                        </xs:complexType>
+                    </xs:schema>
+                    """;
+                // Add microsoft serialization schema subset for dateOnly/timeOnly pattern facets
+                var exporter = new XsdDataContractExporter();
+                exporter.Export(typeof(DateOnly));
+                exporter.Export(typeof(TimeOnly));
+                exporter.Export(typeof(DateTime));
+                XmlSchemaSet set = new XmlSchemaSet();
+                set.Add(XmlSchema.Read(new StringReader(schema), null));
+                foreach (XmlSchema s in exporter.Schemas.Schemas())
+                {
+                    set.Add(s); // contains dateOnly/timeOnly definitions
+                }
+                set.Compile();
+                // Import full set (will pick up the complex type)
+                imp.Import(set);
+                string code = SchemaUtils.DumpCode(imp.CodeCompileUnit);
+                // Should contain DateOnly and TimeOnly members in the generated container type
+                Assert.Contains("class DateOnlyTimeOnlyContainer : object", code);
+                Assert.Contains("System.DateOnly DateOnlyValueField;", code);
+                Assert.Contains("System.TimeOnly TimeOnlyValueField;", code);
+                // This is AI confusion with how xsd:time maps in XmlSerializer. With DCS,
+                // xsd:time maps to - and should continue to map to - String.
```

Comment:
I don't understand the context for mentioning AI confusion here?

Reply:
Mental note when I was going over the test the agent first created. There isn't any crossover between TimeOnly and DateTime as xs:time in the DCS world. Because DCS is very opinionated, a TimeOnly is a TimeOnly and that's that. So this whole 'XsdTimeValueField' and 'XsdDateTimeValueField' aren't really relevant here like they would be in an XmlSerializer-oriented schema test.